ui/src/app/shared/header/header.component.ts

```diff
@@ -279,16 +279,31 @@ export class JobsPaginatorIntl extends MatPaginatorIntl {
   }
 
   getRangeLabel = (page: number, pageSize: number, length: number) => {
-    if (this.backendJobs.value.exhaustive) {
+
+    let knownLength = null;
+
+    if (this.backendJobs.value.totalKnownResults != null) {
+      knownLength = this.backendJobs.value.totalKnownResults;
+    } else if (this.backendJobs.value.exhaustive) {
+      knownLength = length;
+    } else {
+      knownLength = null;
+    }
+
+    if (knownLength != null) {
       // Can't use proper inheritance here, since MatPaginatorIntl only defines
       // properties, rather than class methods.
-      return this.defaultIntl.getRangeLabel(page, pageSize, length);
+
+
+      console.log(page, pageSize, knownLength)
+      return this.defaultIntl.getRangeLabel(page, pageSize, knownLength);
+    } else {
```

Comment:
nit: can just remove the "else" part and reduce indentation, since you're returning above.

Reply:
This is fine to leave as-is, I should have flagged this as an optional comment. Definitely personal preference - for me, it adds more visual noise (indentation) and cognitive load for following the "normal path" through the code. This is mostly coming from a background in Go where this kind of style is preferred. In this case, that argument's not as compelling as there's arguably no "normal path". But possibly slightly better with inversion:

```
if (knownLength == null) {
  ...
}
return this..defaultIntl...
```